codec-http2/src/main/java/io/netty/handler/codec/http2/internal/hpack/HuffmanEncoder.java

```diff
@@ -115,14 +117,78 @@ public void encode(OutputStream out, byte[] data, int off, int len) throws IOExc
      * @param data the string literal to be Huffman encoded
      * @return the number of bytes required to Huffman encode <code>data</code>
      */
-    public int getEncodedLength(byte[] data) {
-        if (data == null) {
-            throw new NullPointerException("data");
+    public int getEncodedLength(CharSequence data) {
+        if (data instanceof AsciiString) {
+            AsciiString string = (AsciiString) data;
+            try {
+                string.forEachByte(encodedLengthProcessor);
+                return encodedLengthProcessor.length();
```

Comment:
we use `encodedLengthProcessor.len` directly below ... consider being consistent and use accessors in both cases or just remove this getter and directly use `len`.